src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -1460,6 +1460,21 @@
     /**
      * A right parenthesis ({@code )}).
      *
+     * <p>For example:</p>
+     * <pre> check()
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * |--METHOD_DEF -> METHOD_DEF
```

Comment:
Javadoc supports using html tags, so `->` is processed like closing tag, it leads to build failures

Instead use special characters like in this [PR](https://github.com/checkstyle/checkstyle/pull/9378/files) 
So that `->` can be replaced with `-&gt;`

Reply:
All the above requested changes were made.